File: back/src/main/java/com/openclassrooms/mddapi/model/Subject.java
package com.openclassrooms.mddapi.model;

import javax.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "Subject")
public class Subject {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "subject_id")
    private Long id;

    private String name;

    private String description;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "subscriber_count")
    private Long subscriberCount;
}

File: front/src/app/core/interfaces/subject.ts
export interface SubjectResponse {
  _embedded: {
    subjectList: [{
      id: number;
      name: string;
      description: string;
      createdAt: string;
      subscriber_count: number;
      _links: {
        subjects: {
          href: string;
        },
        self: {
          href: string;
        }
      }
    }]
  },
  _links: {
    self: {
      href: string;
    }
  }
}

export interface Subject {
  id: number;
  name: string;
  description: string;
  createdAt: string;
  subscriber_count: number;
  _links: {
    subjects: {
      href: string;
    },
    self: {
      href: string;
    }
  }
}
